fastparquet/writer.py

```diff
@@ -571,6 +574,12 @@ def write(filename, data, partitions=[0], encoding="PLAIN",
                     fmd.row_groups.append(rg)
             fmd.num_rows = sum(rg.num_rows for rg in fmd.row_groups)
         else:
+            if write_index or write_index is None and not (
```

Comment:
Can you propose a nicer way to support Pandas indexes?

My general thoughts on this sort of thing is that it's important to leave neat features to the side somewhat ruthlessly while pursuing general usability and first-users.  Grouping and filtering seem like features for the second round of development.  However, it could be that my objective function for what functionality is "critical" is biased towards Dask instead of single-machine use.  These features may have more impact on single machines than I appreciate; I don't think that they're critical for distributed computation, at least not with current users.